harold/plugins/deploy.py

```diff
@@ -171,11 +171,7 @@ class DeployHoldListener(DeployListener):
     def _handle_request(self, request):
         reason = request.args['reason'][0]
         salon_name = request.args['salon_name'][0]
-        salon = yield self.monitor.salons.by_name(salon_name)
```

Comment:
So this has been a learning experience regarding the twisted framework and handling generators and `Deferred` objects.

This fetching here causes problems.  Also it isn't really needed since `self.monitor.hold` basically does the call as well.